app/crud.py

```
@@ -0,0 +1,65 @@
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Error retrieving ToDos: {e}"
+        ) from e
+
+
+async def get_todo(db: AsyncSession, todo_id: int) -> ToDo:
+    try:
+        result = await db.execute(select(ToDo).where(ToDo.id == todo_id))
+        return result.scalar_one()
+    except NoResultFound:
+        raise HTTPException(status_code=404, detail=f"ToDo with ID:{todo_id} not found")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)